CMakeLists.txt

```diff
@@ -106,36 +108,157 @@ set(AZURE_STORAGE_LITE_SOURCE
   src/blob/blob_client_wrapper.cpp
 )
 
-find_package(CURL REQUIRED)
-
-option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" OFF)
-
-find_package(Threads REQUIRED)
-pkg_search_module(UUID REQUIRED uuid)
-
-if (NOT ${USE_OPENSSL})
-  find_package(GnuTLS REQUIRED)
-  set(EXTRA_INCLUDE ${GNUTLS_INCLUDE_DIR})
-  set(EXTRA_LIBRARIES ${GNUTLS_LIBRARIES})
-else()
+if(APPLE)
+  option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" ON)
   add_definitions(-DUSE_OPENSSL)
+  find_package(CURL REQUIRED)
+  # Adding default brew install DIR for openssl
+  set(OPENSSL_ROOT_DIR ${OPENSSL_ROOT_DIR} "/usr/local/opt/openssl/")
   find_package(OpenSSL REQUIRED)
+  include_directories(${PROJECT_SOURCE_DIR}/include ${CURL_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIR} ${EXTRA_INCLUDE})
+  set(EXTRA_LIBRARIES ${EXTRA_LIBRARIES} system)
+  set(CMAKE_MACOSX_RPATH ON)
+elseif (UNIX)
+  option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" OFF)
+  find_package(Threads REQUIRED)
+  find_package(CURL REQUIRED)
+  include(FindPkgConfig)
+  pkg_search_module(UUID REQUIRED uuid)
+  if (NOT ${USE_OPENSSL})
+    find_package(GnuTLS REQUIRED)
+    set(EXTRA_INCLUDE ${GNUTLS_INCLUDE_DIR})
+    set(EXTRA_LIBRARIES ${GNUTLS_LIBRARIES})
+    include_directories(${PROJECT_SOURCE_DIR}/include ${CURL_INCLUDE_DIRS} ${UUID_INCLUDE_DIR} ${EXTRA_INCLUDE})
+  else()
+    add_definitions(-DUSE_OPENSSL)
+    find_package(OpenSSL REQUIRED)
+    include_directories(${PROJECT_SOURCE_DIR}/include ${CURL_INCLUDE_DIRS} ${OPENSSL_INCLUDE_DIR} ${UUID_INCLUDE_DIR} ${EXTRA_INCLUDE})
+  endif()
+else()
+  # Add the parameter/definition specifically for Windows.
+  option(USE_OPENSSL "Use OpenSSL instead of GnuTLS" ON)
+  option(CMAKE_BUILD_TYPE "The default build type" "Debug")
+  option(CMAKE_GENERATOR_PLATFORM "The Generator platform used for CMake" "x86")
+  #Construct the libcurl folder path
+  #This is because the cmake's FindCurl.cmake not being compatible with libcurl's buildscript on Windows.
+  #e.g. libcurl-vc15-x64-debug-static-ssl-static-ipv6-sspi
+  if((DEFINED CURL_ROOT_DIR) AND (NOT (CURL_ROOT_DIR STREQUAL "")))
```

Comment:
ok, this is valid only for Windows? could this be available for Mac and Linux too? There are two reasons why we want to have this:
- We have a custom curl library including: https://github.com/snowflakedb/libsnowflakeclient/blob/master/deps/curl-7.58.0/lib/vtls/sf_ocsp.c
- MacOS curl had a bug that could not enforce TLS 1.2